doc/application_workflow/README.md

```diff
@@ -0,0 +1,56 @@
+
+1. Docker Compose (basic)
+2. Registry pull from docker hub and private registry
+3. Named Data Volumes
+4. Anonymous Data Volumes
+5. Bridged Networks
+6. External networks
+7. Port mapping
+8. Network links/alias
+
+#### Future Features
+
+Be aware the following feature must be taken into account when containerizing an app and deploying it.
+
+1. Docker build
+2. Registry pushing
+3. Data volume sharing between containers
+4. Local host folder mapping to a container volume
+5. Local host file mapping to a container
+6. Docker copy files into a container
+7. Non authenticating registries
+8. Docker container inspect does not return all container network for a container
+
+## Workflow Guidelines
+
+Anything that can be performed with Docker Compose can be performed manually via the Docker CLI and via scripting using the CLI.  This makes Compose a good baseline reference and our guidelines will use it for demonstration purposes.  Our guideline uses Docker Compose 1.8.1.  The list above in the Future Features section puts constraints on what types of containerized application can be deployed on VIC 0.7.0.
+
+Please note, these guidelines and recommendations exist for the current feature set in VIC 0.7.0.  As VIC approaches 1.0, many of these constraints will go away.
+
+#### Guidelines for Building Container Images
+The current lack of docker build and registry pushing means users will need to use regular Docker to build a container and to push it to the global hub or your corporate private registry.  The example workflow using Docker's voting app will illustrate how to get around this constraint.
+
+#### Guidelines for Sharing Config
+VIC 0.7.0 current lack of data volume sharing and docker copy will put constraints on how configuration are provided to a containerized application.  An example of configuration is your web server config files.  Our recommendation for getting around the current limitation is pass in configuration via command line arguments or environment variables.  Both of which are supported by the Docker CLI and Docker Compose.  Add a script to the container image that ingest the command line argument/environment variable and pass these configuration to the contained application.  A benefit of using environment variables to transfer configuration is the containerized app will more closely follow the popular 12-factor app model.
+
+The current lack of local host folder (and file) mapping to container volume limits file sharing between containers.  An example of this is a web server container and another container that contains files that needs the first webserver container to serve.  Another example is two containers sharing a sql lite file.  Our recommendation to work around this current limitation is to build containers that combines all processes that needs to share files between them.  Take for example, if two containers need to share the same sql lite file, combine both container process into one container.
```

Comment:
I'd re-frame this entire paragraph:

```
With no direct support for sharing volumes between containers processes that must share files have the following options:
1. build them into the same image and run in the same container
2. add a script to the container that mounts an NFS share (containers must be on the same network)
   a. Run container with NFS server sharing a data volume
   b. Mount NFS share in whichever containers need to share

<example>
```

NFS client is built into kernel (we should verify with the ESX kernel) so no prereq there. We could provide a docker CLI or compose fragment for starting such a server for illustration.